modules/configuration/src/main/java/org/apache/ignite/internal/configuration/asm/ConfigurationAsmGenerator.java

```diff
@@ -2611,7 +2622,13 @@ private BytecodeBlock treatSourceForConstruct(
         
         return codeBlock;
     }
-    
+
+    private String polyConfTypeIsNotDefinedMessage(Field polymorphicIdField) {
+        return "Polymorphic configuration type is not defined: "
+                + polymorphicIdField.getDeclaringClass().getName()
+                + ". Maybe default value is missing on a @PolymorphicId field (with hasDefault=true)?";
```

Comment:
Maybe the default is not needed, I think you can refer to the documentation for the annotation, and then describe how it should be, 

It is possible not to create a separate method for the message in this case.

Reply:
Why ignore?

Reply:
I actually changed the error message, I just forgot to mention it here. As it is an internal assertion, I changed it a bit differently to help the one who hits it debug the code. I'm not sure whether it makes sense to copy-paste the documentation from an annotation here.

Is this looking better now?